### PR TITLE
fix(scully): replace falsy check with explicit string check to allow root route

### PR DESCRIPTION
**Pull request by BlindDespair**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/scullyio/scully/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #275

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

File: scully/utils/defaultAction.ts [hunk 1]
```diff
@@ -46,7 +46,7 @@ export const generateAll = async (localBaseFilter = baseFilter) => {
     performanceIds.add('Discovery');
     log('Pull in data to create additional routes.');
     const handledRoutes = await addOptionalRoutes(
-      unhandledRoutes.filter((r: string) => r && r.startsWith(localBaseFilter))
+      unhandledRoutes.filter((r: string) => typeof r === 'string' && r.startsWith(localBaseFilter))
```
Comment (BlindDespair) [inline scully/utils/defaultAction.ts]: I am still not sure if this check is needed at all. Maybe it makes sense to just omit it?

Comment (SanderElias) [reply scully/utils/defaultAction.ts]: It protects to mistakes in the Scully.config file. so, yes the check is needed. We have no control over the data that is coming from the guess parser, and neither on the config file. If a route is not a string, its pretty safe to ignore it.

**Comment by SanderElias** [comment]
If you sign the CLA, I can accept this PR.

**Comment by BlindDespair** [comment]
> 
> 
> If you sign the CLA, I can accept this PR.

I signed it now.

File: scully/utils/defaultAction.ts (hunk 1 above)
Comment (BlindDespair) [reply scully/utils/defaultAction.ts]: That makes sense, thank you.

**Pull request merged**